common/retry_delays.go

```diff
@@ -0,0 +1,138 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/agglayer/aggkit/config/types"
+)
+
+var (
+	// If this error is in the returned error, it means that the function should be aborted
+	// and no more retries should be attempted.
+	ErrAbort          = fmt.Errorf("abort")
+	ErrInvalidConfig  = fmt.Errorf("invalid retry delays config")
+	ErrExecutionFails = fmt.Errorf("execution fails after retries exceeded")
+)
+
+// RetryDelays is a struct that holds the retry delays and the maximum number of retries.
+// It implements the RetryDelayer interface, which allows executing a function with retry logic.
+// The delays are specified as a slice of types.Duration, and the maximum number of retries can be set.
+// If maxRetries is set to 0, it means infinite retries are allowed.
+//
+// You can also abort the retrying wrapping the error ErrAbort into the result
+// this is useful if there are some conditions that should not be retried
+type RetryDelays struct {
+	Delays []types.Duration
```

Comment:
Just to be sure these are used in config:

```suggestion
	Delays []types.Duration `mapstructure:"Delays"`
	MaxAttempts int `mapstructure:"MaxAttempts"`
```